fuel-p2p/src/orchestrator.rs

```diff
@@ -93,36 +97,34 @@ impl NetworkOrchestrator {
                     }
                 },
                 p2p_event = p2p_service.next_event() => {
-                    if let FuelP2PEvent::Behaviour(behaviour_event) = p2p_event {
-                        match behaviour_event {
-                            FuelBehaviourEvent::GossipsubMessage { message, .. } => {
-                                match message {
-                                    GossipsubMessage::NewTx(tx) => {
-                                        let _ = self.tx_transaction.send(TransactionBroadcast::NewTransaction(tx));
-                                    },
-                                    GossipsubMessage::NewBlock(block) => {
-                                        let _ = self.tx_block.send(BlockBroadcast::NewBlock(block));
-                                    },
-                                    GossipsubMessage::ConsensusVote(vote) => {
-                                        let _ = self.tx_consensus.send(ConsensusBroadcast::NewVote(vote));
-                                    },
+                    match p2p_event {
+                        FuelP2PEvent::GossipsubMessage { message, .. } => {
+                            match message {
+                                GossipsubMessage::NewTx(tx) => {
+                                    let _ = self.tx_transaction.send(TransactionBroadcast::NewTransaction(tx));
+                                },
+                                GossipsubMessage::NewBlock(block) => {
+                                    let _ = self.tx_block.send(BlockBroadcast::NewBlock(block));
+                                },
+                                GossipsubMessage::ConsensusVote(vote) => {
+                                    let _ = self.tx_consensus.send(ConsensusBroadcast::NewVote(vote));
+                                },
+                            }
+                        },
+                        FuelP2PEvent::RequestMessage { request_message, request_id } => {
+                            match request_message {
+                                RequestMessage::RequestBlock(block_height) => {
+                                    let db = self.db.clone();
+                                    let tx_outbound_response = self.tx_outbound_responses.clone();
+
+                                    tokio::spawn(async move {
```

Comment:
[nit] I know it's not the most common occurrence, and this already existed before (so it may warrant an issue if you think it's worth changing) but should this thread have some handler on it? So in the event the node has the chance to gracefully shutdown the peer can get their block rather than timing out

Reply:
Agree this probably requires an issue but this is not an ideal pattern as the spawned tasks act like an unbounded channel in this case. If Requests come in faster then responses then your upper bound is the computers memory because you have no back pressure.

Reply:
good point, we can do that in another PR, don't want to touch it now since it's not related to this PR + there might be some changes how the orchestrator works because of the design changes we are introducing